File: server/src/main/java/org/dan/jadalnia/app/ws/MessageForClient.java
package org.dan.jadalnia.app.ws;

import com.fasterxml.jackson.annotation.JsonSubTypes;
import com.fasterxml.jackson.annotation.JsonTypeInfo;

@JsonTypeInfo(
        use = JsonTypeInfo.Id.NAME,
        include = JsonTypeInfo.As.PROPERTY)
@JsonSubTypes({
        @JsonSubTypes.Type(value = PropertyUpdated.class, name = "propertyUpdated")
})
public interface MessageForClient {
}

File: service/src/main/kotlin/org/dan/jadalnia/app/ws/MessageForClient.kt
package org.dan.jadalnia.app.ws

import com.fasterxml.jackson.annotation.JsonSubTypes
import com.fasterxml.jackson.annotation.JsonTypeInfo
import org.dan.jadalnia.app.order.OrderExecutingEvent
import org.dan.jadalnia.app.order.OrderStateEvent
import org.dan.jadalnia.app.token.TokenApprovedEvent

@JsonTypeInfo(
        use = JsonTypeInfo.Id.NAME,
        include = JsonTypeInfo.As.PROPERTY)
@JsonSubTypes(value = [
    JsonSubTypes.Type(
            value = PropertyUpdated::class,
            name = "propertyUpdated"),
    JsonSubTypes.Type(
        value = OrderExecutingEvent::class,
        name = "orderExecuting"),
    JsonSubTypes.Type(
        value = OrderStateEvent::class,
        name = "orderState"),
    JsonSubTypes.Type(
        value = TokenApprovedEvent::class,
        name = "tokenApproved")
])
interface MessageForClient
